x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/components/steps/components/package_policy_input_stream.tsx

```diff
@@ -90,12 +95,24 @@ export const PackagePolicyInputStreamConfig = memo<Props>(
       !!packagePolicyInputStream.id &&
       packagePolicyInputStream.id === defaultDataStreamId;
     const isPackagePolicyEdit = !!packagePolicyId;
-    const isInputOnlyPackage = packageInfo.type === 'input';
 
-    const hasDatasetVar = packageInputStream.vars?.some(
-      (varDef) => varDef.name === DATASET_VAR_NAME
-    );
-    const showPipelinesAndMappings = !isInputOnlyPackage && !hasDatasetVar;
+    const customDatasetVar = packagePolicyInputStream.vars?.[DATASET_VAR_NAME];
+    const customDatasetVarValue = customDatasetVar?.value?.dataset || customDatasetVar?.value;
+
+    const { exists: indexTemplateExists, isLoading: isLoadingIndexTemplate } =
+      useIndexTemplateExists(
+        getRegistryDataStreamAssetBaseName({
+          dataset: customDatasetVarValue,
+          type: packageInputStream.data_stream.type,
+        })
+      );
+
+    // only show pipelines and mappings if the matching index template exists
+    // in the legacy case (e.g logs package pre 2.0.0) the index template will not exist
+    // because we allowed dataset to be customized but didnt create a matching index template
+    // for the new dataset.
+    const showPipelinesAndMappings = !isLoadingIndexTemplate && indexTemplateExists;
```

Comment:
Would it be possible to add a test anywhere to verify these components are shown when `type === 'input'` for a given package?

Reply:
Added cypress tests 👍